core/src/com/unciv/logic/map/InfluenceMap.kt

```diff
@@ -0,0 +1,206 @@
+package com.unciv.logic.map
+
+import com.unciv.Constants
+import com.unciv.logic.battle.CityCombatant
+import com.unciv.logic.civilization.CivilizationInfo
+
+object InfluenceMap {
+
+    fun setInfluenceScore(viewingCiv: CivilizationInfo, tilesToCheck: List<TileInfo>? = null) {
+        val viewable = viewingCiv.gameInfo.tileMap.values
+        
+        //Such a dirty hack, need to actually create an object that holds these for each Civ
+        for (tile in viewable) {
+            tile.enemyInfluenceScore = 0.0F
+            tile.friendlyInfluenceScore = 0.0F
+        }
+        val tilesWithEnemies = (tilesToCheck ?: viewable)
+                .filter { containsEnemyUnit(it, viewingCiv) }
+        val tilesWithNeutralCombatant = (tilesToCheck ?: viewable)
+                .filter { containsNeutralUnit(it, viewingCiv) }
+        val tilesWithFriendlyCombatant = (tilesToCheck ?: viewable)
+                .filter { containsFriendlyUnit(it, viewingCiv) }
+        setEnemyTiles(tilesWithEnemies)
+        setNeutralTiles(tilesWithNeutralCombatant)
+        setFriendlyTiles(tilesWithFriendlyCombatant)
+
+    }
+    
+    private fun setEnemyTiles(enemyTiles: List<TileInfo>) {
+        for (enemyTile in enemyTiles) {
+            for (unit in enemyTile.getUnits()) {
+                if (unit.type.isMilitary()) {
+                    enemyTile.enemyInfluenceScore += (unit.baseUnit.strength * (1 + enemyTile.getDefensiveBonus())) * (unit.health/100.0F)
```

Comment:
The tile and health modifier is something you use several times in the class, you should probably implement a getModifiedUnitStrength function